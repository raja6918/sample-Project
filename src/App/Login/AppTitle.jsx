import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const AppTitleContainerWrapper = styled.div`
  & h2 {
    font-size: 36px;
    color: #ffffff;
    margin-top: 0;
    margin-bottom: 5px;
    font-weight: 400;
  }
  & p {
    font-size: 20px;
    font-weight: 200;
    color: #ffffff;
    margin: 0;
    opacity: 0.75;
  }
`;

// eslint-disable-next-line
const shortHash = COMMITHASH.substr(0, 7);

const AppTitleContainer = ({ t }) => {
  return (
    <AppTitleContainerWrapper>
      <h2>{t('LOGIN.appTitle')}</h2>
      <p title={`Build ${shortHash}`}>{t('LOGIN.appVersion')}</p>
    </AppTitleContainerWrapper>
  );
};

AppTitleContainer.propTypes = {
  t: PropTypes.func.isRequired,
};

export default AppTitleContainer;
