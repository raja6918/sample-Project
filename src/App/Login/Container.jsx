import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const LoginContainerWrapper = styled.div`
  background-image: url(${props => props.bg});
  background-size: cover;
  background-position: center;
  height: 100vh;
  background-repeat: no-repeat;
  position: relative;
`;

const LoginContainer = ({ children, t }) => {
  let bg = '';
  try {
    bg = require(`./${t('LOGIN.bg')}`);
  } catch (e) {
    bg = require(`./Login.jpg`);
  }

  return <LoginContainerWrapper bg={bg}>{children}</LoginContainerWrapper>;
};

LoginContainer.propTypes = {
  children: PropTypes.element,
  t: PropTypes.func.isRequired,
};

LoginContainer.defaultProps = {
  children: <div />,
};

export default LoginContainer;
