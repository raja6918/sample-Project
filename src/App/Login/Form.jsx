import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import MUIButton from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';

const LoginBox = styled.div`
  width: 425px;
  height: 330px;
  padding: 40px;
  background-color: #ffffff;
  z-index: 999;
  box-shadow: 0 1px 4px 0 rgba(0, 0, 0, 0.5);
  border-radius: 4px;
  margin-top: 64px;
  & form p:first-of-type {
    margin: 0;
    font-weight: 400;
    font-size: 24px;
    color: #ff650c;
    margin-bottom: 20px;
  }
  & form div label {
    color: rgba(0, 0, 0, 0.67);
  }
  & form div label + div::before {
    background-color: transparent;
  }
  & form div:first-child {
    margin-bottom: 13px;
  }
`;

const Input = styled(TextField)`
  width: 100%;
  margin-bottom: ${props => props.mb};
  & div::before {
    background-color: rgba(0, 0, 0, 0.12);
  }
`;

const Button = styled(MUIButton)`
  float: right;
  min-width: 80px;
  max-width: 80px;
  max-height: 32px;
  min-height: 32px;
  background: #0a75c2;
  border-radius: 2px;
  &:hover {
    background: #075187;
  }
  & span {
    font-weight: 400;
    font-size: 13px;
    letter-spacing: 0;
    text-transform: none;
    color: #ffffff;
    line-height: 1.4em;
    opacity: 0.85;
  }
`;

const ErrorMessage = styled.p`
  color: #ff650c;
  height: 38px;
  margin-top: 0;
  margin-bottom: 6px;
  font-size: 13px;
  font-weight: 500;
`;

class Form extends Component {
  handleSubmit = e => {
    e.preventDefault();
    this.props.handleSubmit(this.ref);
  };

  render() {
    const { t, errorMessage, clearErrorMessage } = this.props;

    return (
      <LoginBox>
        <form ref={ref => (this.ref = ref)} onSubmit={this.handleSubmit}>
          <p>{t('LOGIN.loginTitle')}</p>
          <Input
            label={t('LOGIN.username')}
            id="username"
            name="username"
            mb="13px"
            inputProps={{
              maxLength: 50,
            }}
            onChange={clearErrorMessage}
          />
          <Input
            id="password"
            name="password"
            label={t('LOGIN.password')}
            type="password"
            mb="13px"
            inputProps={{
              maxLength: 100,
            }}
            onChange={clearErrorMessage}
          />
          <ErrorMessage>{errorMessage}</ErrorMessage>

          <Button type="submit" variant="contained">
            {t('LOGIN.button')}
          </Button>
        </form>
      </LoginBox>
    );
  }
}

Form.propTypes = {
  t: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  errorMessage: PropTypes.string.isRequired,
  clearErrorMessage: PropTypes.func.isRequired,
};

export default Form;
