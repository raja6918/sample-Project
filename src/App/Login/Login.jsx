import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';
import styled from 'styled-components';

import storage from '../../utils/storage';
import Container from './Container';
import AppTitle from './AppTitle';
import Form from './Form';

import API from '../../utils/API';

import scenarioService from '../../services/Scenarios/index';
import { setUserDetails } from '../../actions/user';
import { connect } from 'react-redux';

const Overlay = styled.div`
  width: 437px;
  background-image: linear-gradient(
    0deg,
    rgba(0, 76, 140, 0.64) 2%,
    #00355b 48%
  );
  border-radius: 4px;
  height: calc(100vh - 80px);
  left: 10%;
  top: 40px;
  position: relative;
  padding: 35px 33px 0 49px;
`;

class Login extends Component {
  state = {
    errorMessage: '',
  };

  redirectToDashboard = () => {
    this.props.history.push('/');
  };

  handleSubmit = ref => {
    const loginRequestBody = {
      username: ref.username.value.trim(),
      password: ref.password.value.trim(),
    };

    const { setUserDetails } = this.props;
    API.post('/auth/login', loginRequestBody)
      .then(response => {
        storage.setItem('jwt', response);
        API.get(`/users/${loginRequestBody.username}`).then(res => {
          storage.setItem('loggedUser', res);
          setUserDetails(res);
          scenarioService.deletePendingPreviews();
          this.redirectToDashboard();
        });
      })
      .catch(error => {
        let errorMessage;
        try {
          const [{ errorDetails }] = error.response.data;
          const [errorInfo] = errorDetails;
          errorMessage = this.props.t(`ERRORS.${errorInfo.messageKey}`);
        } catch (err) {
          if (error.response) {
            if (error.response.status === 401) {
              errorMessage = this.props.t('LOGIN.invalidCredentials');
            } else {
              errorMessage = this.props.t('GLOBAL.serviceNotAvailable');
            }
          } else {
            errorMessage = this.props.t('GLOBAL.serviceNotAvailable');
          }
        }
        this.setState({ errorMessage });
      });
  };

  clearErrorMessage = () => {
    this.setState({ errorMessage: '' });
  };

  render() {
    const { t } = this.props;
    const { errorMessage } = this.state;

    return (
      <React.Fragment>
        <Container t={t}>
          <Overlay>
            <AppTitle t={t} />
            <Form
              t={t}
              handleSubmit={this.handleSubmit}
              errorMessage={errorMessage}
              clearErrorMessage={this.clearErrorMessage}
            />
          </Overlay>
        </Container>
      </React.Fragment>
    );
  }
}

Login.propTypes = {
  t: PropTypes.func.isRequired,
  history: PropTypes.shape({
    location: PropTypes.shape({
      pathname: PropTypes.string,
    }),
    push: PropTypes.func,
  }).isRequired,
  setUserDetails: PropTypes.func.isRequired,
};

const mapStatetoProps = state => {
  return { userData: state.user.userData };
};

const mapDispatchToProps = dispatch => {
  return {
    setUserDetails: data => dispatch(setUserDetails(data)),
  };
};
const LoginComponent = connect(
  mapStatetoProps,
  mapDispatchToProps
)(Login);
export default translate()(LoginComponent);
