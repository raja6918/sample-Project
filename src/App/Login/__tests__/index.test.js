import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';

import Login from '../index';

const props = {
  t: jest.fn(),
  history: {},
};

test('Login Component renders', () => {
  const wrapper = shallow(
    <Router initialEntries={['/testing']} initialIndex={0}>
      <Login {...props} />
    </Router>
  );
  expect(wrapper.instance().history.location.pathname).toBe('/testing');
  wrapper.instance().history.push('/');
  expect(wrapper.instance().history.location.pathname).toBe('/');
});

test('Snapshot is up to date', () => {
  const wrapper = shallow(<Login {...props} />);
  expect(wrapper).toMatchSnapshot();
});
