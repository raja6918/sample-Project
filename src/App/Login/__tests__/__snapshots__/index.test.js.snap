// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot is up to date 1`] = `
<Connect(Login)
  history={Object {}}
  t={[MockFunction]}
/>
`;
